addons/mail/static/src/components/chatter_container/chatter_container.js

```diff
@@ -25,25 +25,23 @@ class ChatterContainer extends Component {
     constructor(...args) {
         super(...args);
         this.chatter = undefined;
-        this._wasMessagingInitialized = false;
         useStore(props => {
-            const isMessagingInitialized = this.env.isMessagingInitialized();
-            if (!this._wasMessagingInitialized && isMessagingInitialized) {
-                this._wasMessagingInitialized = true;
-                this.chatter = this.env.models['mail.chatter'].create(props);
-            }
-            return { isMessagingInitialized };
+            this._insertFromProps(props);
+            return { chatter: this.chatter };
         });
     }
 
-    mounted() {
-        this._update();
-    }
-
-    patched() {
-        this._update();
+    /**
+     * @override
+     */
+    willUpdateProps(nextProps) {
+        this._insertFromProps(nextProps);
+        return super.willUpdateProps(...arguments);
```

Comment:
What's the motivation to use `willUpdateProps` instead of `mounted()` & `patched()`?

Reply:
The goal is to update the chatter fields when the props of the container changes. The proper hook for that is `willUpdateProps`, there is no reason to bind on render cycle (it worked because rendering is a consequence of changing props, but we shouldn't rely on that).